bodhi/tests/server/services/test_releases.py

```diff
@@ -480,34 +484,120 @@ def test_get_single_release_html_two_same_updates_same_month(self):
         self.assertIn('f17-updates-testing', res)
         # Since the updates are the same type and from the same month, we should see a count of 2 in
         # the graph data.
-        graph_data = 'data : [\n                2,\n              ]'
+        graph_data = 'data : [\n            2,\n          ]'
+        print(res)
         self.assertTrue(graph_data in res)
 
     def test_get_non_existent_release_html(self):
         self.app.get('/releases/x', headers={'Accept': 'text/html'}, status=404)
 
-    def test_get_releases_html(self):
-        res = self.app.get('/releases/', headers={'Accept': 'text/html'})
-        self.assertEqual(res.content_type, 'text/html')
-        self.assertIn('Fedora 22', res)
-
-    def test_query_releases_html_two_releases_same_state(self):
-        """Test query_releases_html() with two releases in the same state."""
-        attrs = {
-            "name": "F42", "long_name": "Fedora 42", "version": "42",
-            "id_prefix": "FEDORA", "branch": "f42", "dist_tag": "f42",
-            "stable_tag": "f42-updates",
-            "testing_tag": "f42-updates-testing",
-            "candidate_tag": "f42-updates-candidate",
-            "pending_stable_tag": "f42-updates-pending",
-            "pending_signing_tag": "f42-updates-testing-signing",
-            "pending_testing_tag": "f42-updates-testing-pending",
-            "override_tag": "f42-override",
-            "csrf_token": self.get_csrf_token()}
-        self.app.post("/releases/", attrs, status=200)
 
-        res = self.app.get('/releases/', headers={'Accept': 'text/html'})
+class TestReleasesHTML(base.BaseTestCase):
+    def setUp(self, *args, **kwargs):
+        super(TestReleasesHTML, self).setUp(*args, **kwargs)
+
+        def _add_updates(updateslist, user, release, packagesuffix):
+            """Private method that adds updates to the database for testing
+
+            """
+            count = 0
+            for i in updateslist:
+                for j in i[1]:
+                    for k in range(0, j[1]):
+                        update = Update(
+                            user=user,
+                            status=i[0],
+                            type=j[0],
+                            notes='Useful details!',
+                            release=release,
+                            date_submitted=datetime(1984, 11, 2),
+                            requirements='rpmlint',
+                            stable_karma=3,
+                            unstable_karma=-3,
+                        )
+                        self.db.add(update)
+                        self.db.flush()
+                        count = count + 1
```

Comment:
This function is creating Updates, there's a good chance that this is causing the timeout because creating updates now sends out a Fedora Messaging message, and it needs to be wrapped with mock_sends in the unit tests or the code will just wait a long time to connect to a RabbitMQ server. All this delay piles up and usually causes a timeout.

Reply:
Thanks, updated [commit](https://github.com/fedora-infra/bodhi/pull/3500/commits/2e45e91f18ca0609cc1b3e327434bfdabcecb91d) to do this (hopfully correctly)